lib/kdump_utils.pm

```diff
@@ -26,12 +26,17 @@ sub install_kernel_debuginfo {
     zypper_call("-v in $debuginfo", timeout => 4000);
 }
 
+sub get_repo_url_for_kdump_sle {
+    (get_var('REPO_SLES_DEBUG') || get_var('REPO_SLE15_MODULE_BASESYSTEM_DEBUG'))
+      ?
+      join('/', $utils::OPENQA_FTP_URL, get_var(check_var('REPO_SLES_DEBUG') ? 'REPO_SLES_DEBUG' : 'REPO_SLE15_MODULE_BASESYSTEM_DEBUG'))
```

Comment:
This is hard to read. On top, check_var verifies that value equals to second parameter. In this case you don't pass parameter at all, so I guess check_var will compare variable value to undef. I suggest that we split it to two return statements. return %url% if get_var('REPO_SLES_DEBUG'); and return %url% if get_var('REPO_SLE15_MODULE_BASESYSTEM_DEBUG');

Reply:
check_var worked fine with 1 param because https://github.com/os-autoinst/os-autoinst/blob/master/testapi.pm#L647 so I was lucky in the verification, because it would not work with sle12sp4. Good that you noticed.
On the other hand, I want to keep this code elegant and doing proper use of the functions:
Basically it is read like: if there is some variable related with repo try to join a string concatenating one or the other (is exclusive so you don't have in a test both because these are different products) and the final undef of the subroutine is to give to calling function a way to check. I was using these two-if approach at the beginning and I finally got really more satisfied with this one specially after the tidy command split it in several lines.
Verification run: http://dhcp227/tests/58

Reply:
You still can compare version and matching variable, like 

``` 
return ... if sle15 and get_var(...)
return ... if get_var(...)
```

I don't think we can find cute one-liner to solve our problem, but this is really hard to read, especially with this formatting. But you will end up with 2 lines instead of 4. Additionally, here we check variables irregardless the version, so if both of them will be defined, we may use wrong one. It's fine as we assume this won't happen, but possibility is there. If we would check versions we could use get_required_var and throw relevant error as test doesn't make much sense without debug repo.

Reply:
aha, I think now I understand your point, because in perl if I don't return anything will be ok, it will be undef, so I don't need the logic for the undef, but we need to be careful with this kind of thing for maintenance. Let me change it.